build-tools/packages/build-cli/src/handlers/askFunctions.ts

```diff
@@ -0,0 +1,75 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "assert";
+import { bumpVersionScheme } from "@fluid-tools/version-tools";
+import inquirer from "inquirer";
+import { Machine } from "jssm";
+import { getDefaultBumpTypeForBranch, getPreReleaseDependencies } from "../lib";
+import { CommandLogger } from "../logging";
+import { MachineState } from "../machines";
+import { FluidReleaseStateHandlerData } from "./fluidReleaseStateHandler";
+import { BaseStateHandler, StateHandlerFunction } from "./stateHandlers";
+
+/**
+ * Determines the release type based on context, or by asking the user if needed.
+ *
+ * @param state - The current state machine state.
+ * @param machine - The state machine.
+ * @param testMode - Set to true to run function in test mode.
+ * @param log - A logger that the function can use for logging.
+ * @param data - An object with handler-specific contextual data.
+ * @returns True if the state was handled; false otherwise.
```

Comment:
@Josmithr Do you know of a better way to handle this boilerplate? The arguments and descriptions are the same for all the functions.

Reply:
You could define and accept some base props interface or something. If specific handlers need additional arguments, they can extend the base interface.

Reply:
I was hoping there might be some magic tsdoc syntax or something. Oh well. I'm not a huge fan of the "props" model. :( I sort of use it here with the `data` param, which is `unknown` in the base classes but has a concrete type specific to the "handler type" (hand-wavey concept) in the class implementations. So wrapping those props into an object makes sense for that use case.

Wrapping _all_ the function args in an object always feels wrong to me for some reason. Maybe it's the added level of indirection? Actually, I think my attitude stems from using JS projects in the past that use this model and it was always really hard to tell what arguments a function actually took. JS being untyped made this impossible without consulting the docs. At least with function args you have a name that gives you an idea of what to pass. When it's an object you lose those contextual clues. But types solve most of those problems, so maybe I should get over it.

I'll think about this some more.